src/cdomains/addressDomain.ml

```diff
@@ -133,7 +134,7 @@ struct
     | false, false -> cop x y
 
   let meet x y   = merge join meet x y
-  let narrow x y = merge widen narrow x y
+  let narrow x y = merge (fun x y -> (join x y) |> widen x) narrow x y
```

Comment:
You beat me to it opening the PR, but this is indeed where the invalid call is. I would format it like how it's done elsewhere (e.g. solvers):
```suggestion
  let narrow x y = merge (fun x y -> widen x (join x y)) narrow x y
```